File: lab0_helloworld/node_demo.cpp

#include <rclcpp/rclcpp.hpp>

int main(int argc, char *argv[]) {

    rclcpp::init(argc, argv);
    
    std::string node_name = "demo_cpp_node";
    auto node = std::make_shared<rclcpp::Node>(node_name);
    auto logger = node->get_logger();
    RCLCPP_INFO(logger, "hello from %s", node_name.c_str());

    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: lab0_helloworld/node_demo.py

import rclpy

from rclpy.node import Node

def main(args=None):
    rclpy.init()
    node_name = "demo_py_node"
    node = Node(node_name)
    _logger = node.get_logger()
    _logger.info(f"Hello from {node_name}")
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()
